src/lib/stable/ic_apis/is_controller.ts

```diff
@@ -21,8 +21,10 @@ export function isController(principal: Principal): boolean {
     }
 
     if (globalThis._azleIcExperimental !== undefined) {
+        const principalBytes =
+            principal.toUint8Array() as Uint8Array<ArrayBuffer>;
```

Comment:
Whoa let's discuss this, I think there's a better way than these casts, I hope

Reply:
Okay let's discuss it